src/main/treeDataProviders/dependenciesTree/dependenciesRoot/pnpmTree.ts

```diff
@@ -0,0 +1,76 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { LogManager } from '../../../log/logManager';
+import { GeneralInfo } from '../../../types/generalInfo';
+import { PackageType } from '../../../types/projectType';
+import { NpmUtils, ProjectDetails } from '../../../utils/npmUtils';
+import { PnpmUtils } from '../../../utils/pnpmUtils';
+import { DependenciesTreeNode } from '../dependenciesTreeNode';
+import { BuildTreeErrorType, RootNode } from './rootTree';
+
+export class PnpmTreeNode extends RootNode {
+    private static readonly COMPONENT_PREFIX: string = 'npm://';
+
+    constructor(fullPath: string, private _logManager: LogManager, parent?: DependenciesTreeNode) {
+        super(fullPath, PackageType.Pnpm, parent);
+    }
+
+    public async refreshDependencies() {
+        let results: any;
+        try {
+            results = PnpmUtils.runPnpmLs(this.workspaceFolder);
+            this.processProjectsInformation(results);
+        } catch (error) {
+            this._logManager.logError(<any>error, false);
+            this._logManager.logMessageAndToastErr(
+                `Failed to scan pnpm project. Hint: Please make sure the commands 'pnpm install' run successfully in '${this.workspaceFolder}'`,
+                'ERR'
+            );
+            this.buildError = BuildTreeErrorType.NotInstalled;
+        }
+
+        const details: ProjectDetails = NpmUtils.getProjectDetailsFromPackageJson(this.workspaceFolder);
+        this.generalInfo = new GeneralInfo(
+            details.projectName,
+            details.projectVersion,
+            [],
+            path.join(this.workspaceFolder, 'pnpm-lock.yaml'),
```

Comment:
No, `this.fullPath` is pointing to the `package.json` because we want to show it in the result tree as the descriptor.
We use the information in the GeneralInfo to make sure the indicator is marked as parsed when processing the package manager descriptors.

I added a comment about it